File: codigos-probar-temas/codigo-prueba-temas.js
// Función simple que suma dos números
function sumar(a, b) {
  return a + b;
}

// Clase Persona con constructor y método
class Persona {
  constructor(nombre, edad) {
    this.nombre = nombre;
    this.edad = edad;
  }

  saludar() {
    console.log(`Hola, soy ${this.nombre} y tengo ${this.edad} años.`);
  }
}

// Uso de la clase Persona
let persona = new Persona("Juan", 30);
persona.saludar();

// Condicional simple
let num1 = 5,
  num2 = 10;
if (num1 < num2) {
  console.log(`${num1} es menor que ${num2}`);
}

// Bucle for
for (let i = 0; i < 5; i++) {
  console.log(`Iteración ${i}`);
}

// Manejo de promesas (asíncrono)
function esperar(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

async function pruebaAsync() {
  console.log("Esperando 1 segundo...");
  await esperar(1000);
  console.log("Hecho.");
}

pruebaAsync();

File: codigos-probar-temas/codigo-prueba-temas.c
#include <stdio.h>

// Función simple que suma dos números
int sumar(int a, int b) { return a + b; }

// Estructura de datos (struct) para probar
struct Persona {
    char nombre[50];
    int edad;
};

// Función que imprime los datos de una persona
void imprimir_persona(struct Persona p) {
    printf("Nombre: %s\n", p.nombre);
    printf("Edad: %d\n", p.edad);
}

int main() {
    // Variables básicas
    int num1 = 5, num2 = 10;
    printf("La suma de %d y %d es: %d\n", num1, num2, sumar(num1, num2));

    // Condicional simple
    if (num1 < num2) {
        printf("%d es menor que %d\n", num1, num2);
    }

    // Bucle for
    for (int i = 0; i < 5; i++) {
        printf("Iteración %d\n", i);
    }

    // Crear una persona y mostrar sus datos
    struct Persona persona = {"Juan", 30};
    imprimir_persona(persona);

    return 0;
}
